### ISSUE TITLE
Support additional fields for GKE

### ISSUE THREAD

**Issue by akshay196** [kubernetes-sigs/cluster-api-provider-gcp#906]
/kind feature

**Describe the solution you'd like**

CAPG initial support for Managed Cluster (GKE) has added few basic configurations. GCP does support lot more fields for example private cluster, enable workload identity, just to name a few. This is a request to add support for more fields than just basics.

[This spreadsheet](https://docs.google.com/spreadsheets/d/1ETX0JyIL5jiOWPUrxOSOJ8yCzXvSERp2bZ3jPn33Vdo/edit#gid=0) enlists all important GKE  configuration we can support in GCPManaged* kinds.

All additional GKE fields are documented in the spreadsheet linked above. The structure in spreadsheet can be used for defining GCPManaged API specs. Feel free to comment to discuss more about any fields.

**Anything else you would like to add:**

There are other open issues those requested for additional fields too:
- https://github.com/kubernetes-sigs/cluster-api-provider-gcp/issues/903
- https://github.com/kubernetes-sigs/cluster-api-provider-gcp/issues/897

Such requests can be tackled by the complete list (spreadsheet) attached here.

If this sounds good then, I am happy to take this forward and decide sub-tasks for better collaboration.

looping @poornima-rafay @niravparikh05

**Progress**

```[tasklist]
### Tasks - PR1
- [x] GCPManagedMachinePool changes (@akshay196) - https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/944
```

```[tasklist]
### Tasks - PR2
- [ ] GCPManagedControlPlane changes - part 1 (Description, ClusterNetwork, DefaultNodeLocation, DefaultMaxPodsPerNode) - ~`PR#936` by Anniruddha~ (@akshay196) - https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/1186
```

```[tasklist]
### Tasks - PR3
- [ ] GCPManagedControlPlane changes - part 2 (ClusterSecurity, AddonsConfig, LoggingConfig, MonitoringConfig) (@akshay196) - https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/1186
```

```[tasklist]
### Tasks - PR4
- [ ] Update relevant documentation if any
```

**Comment by aniruddha2000** [comment]
@akshay196 I am happy to get involved in the dividing task discussion and take some of them.

**Comment by richardcase** [comment]
That spreadsheet is really helpful @akshay196 :+1: 

It would be good to split the work into smaller chunks to make the review easier and also the addition of e2e tests.

**Comment by akshay196** [comment]
Updated spreadsheet with more details. Added task list above.

**Comment by aniruddha2000** [comment]
Thanks for the update @akshay196 , have you started working on anything? If not I will start adding fields one by one and open a tracking PR.

**Comment by akshay196** [comment]
@aniruddha2000 Started adding username in tasks list for taking ownership of tasks.

**Comment by aniruddha2000** [comment]
@akshay196 I have created the PR for my first task https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/936 feel free to link it with the description tasks. Also, could you have a look into it? I officially cannot make you a reviewer because you are not in the k8s-sig org.

**Comment by aniruddha2000** [comment]
I have started looking in the Reconciliation loop of the same, could you assign me the `reconciliation loop for GCPManagedControlPlane` @akshay196 ?

**Comment by jayesh-srivastava** [comment]
Hi @aniruddha2000 @akshay196 If there's room for more, would like to contribute as well.

**Comment by aniruddha2000** [comment]
@jayesh-srivastava Sure, you can look into the webhooks part maybe, I think we need to discuss what kind of webhooks are required after all of the fields are up.

**Comment by jayesh-srivastava** [comment]
Sounds good @aniruddha2000. @akshay196 Can you please update the description and assign the webhook tasks to me?

**Comment by sayantani11** [comment]
Hello @aniruddha2000 @jayesh-srivastava long time no see!!! Would also like to help in the implementation!!! Lemme know if anything is there.

**Comment by sayantani11** [comment]
Hello @aniruddha2000 @jayesh-srivastava long time no see!!! Would also like to help in the implementation!!! Lemme know if anything is there.

**Comment by aniruddha2000** [comment]
@sayantani11 yes definitely, you can have a look at other webhook

**Comment by liticer** [comment]
@aniruddha2000 @jayesh-srivastava  How are the latest developments? We want to use these features, and if they could not be ready soon, we would like to build further implementations based on the current code.

**Comment by akshay196** [comment]
Updated tasks above as per discussion [here](https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/944#issuecomment-1682160157). We have divided GCPManagedControlPlane additional fields into two. We will raise PR that includes set of new fields, respective reconciliation loop changes, webhooks and e2e if any.
cc @aniruddha2000 You will need to update your PR.

Task 3 is up for anyone interested.
